apps/server/doc/testing.md

```diff
@@ -177,20 +216,27 @@ beforeEach(async () => {
 	mockService = module.get(MockService);
 });
 
-afterEach(async () => {
+afterAll(async () => {
 	await module.close();
 });
+
+afterEach(() => {
+	jest.resetAllMocks();
+})
 ```
+`createTestingModule` should only be calld in `beforeAll` and not in `beforeEach` to keep the setup and teardown for each test as simple as possible. Therefore `.close` methods should only be called in `afterAll` and not in `afterEach`.
 
-the resulting mock has all the functions of the original `Class`, replaced with jest spies. This gives you the full intellij features of the original class including code completion and typesavety, combined with all the features of spies.
+To generally reset specific mock implementation after each test `jest.resetAllMocks` can be used in afterEach. `jest.restoreAllMocks` should not be used, because in some cases it will not properly restore mocks created by ts-jest.
+
+The resulting mock has all the functions of the original `Class`, replaced with jest spies. This gives you the full intellij features of the original class including code completion and typesavety, combined with all the features of spies.
 
 ```Typescript
 describe('somefunction', () => {
 	describe('when service returns user', () => {
 		const setup = () => {
 			const resultUser = userFactory.buildWithId();
 
-			mockService.getUser.mockReturnValue(resultUser);
+			mockService.getUser.mockReturnValueOnce(resultUser);
```

Comment:
Ich wurde das nicht so gerne festlegen, mach in solche Fälle mehr Schreiberei, meine Meinung nach.